src/main/java/ch/njol/skript/expressions/ExprXOf.java

```diff
@@ -45,75 +43,53 @@
 @Examples("give level of player of pickaxes to the player")
 @Since("1.2")
 public class ExprXOf extends PropertyExpression<Object, Object> {
+
 	static {
-		Skript.registerExpression(ExprXOf.class, Object.class, ExpressionType.PATTERN_MATCHES_EVERYTHING, "%number% of %itemstacks/entitytype%");
+		Skript.registerExpression(ExprXOf.class, Object.class, ExpressionType.PATTERN_MATCHES_EVERYTHING, "%number% of %itemstacks/itemtypes/entitytype%");
 	}
-	
-	@SuppressWarnings("null")
-	Expression<Number> amount;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
+	private Expression<Number> amount;
+
+	@SuppressWarnings("unchecked")
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr(exprs[1]);
 		amount = (Expression<Number>) exprs[0];
-		if (amount instanceof Literal && getExpr() instanceof Literal)// "x of y" is also an ItemType syntax
-			return false;
-		return true;
+		// "x of y" is also an ItemType syntax
+		return !(amount instanceof Literal) || !(getExpr() instanceof Literal);
 	}
-	
-	@Override
-	public Class<? extends Object> getReturnType() {
-		return getExpr().getReturnType();
-	}
-	
+
 	@Override
-	protected Object[] get(final Event e, final Object[] source) {
-		return get(source, new Converter<Object, Object>() {
-			@Override
-			@Nullable
-			public Object convert(final Object o) {
-				final Number a = amount.getSingle(e);
-				if (a == null)
-					return null;
-				if (o instanceof ItemStack) {
-					final ItemStack is = ((ItemStack) o).clone();
-					is.setAmount(a.intValue());
-					return is;
-				} else if (o instanceof ItemType) {
-					ItemType type = ((ItemType) o).clone();
-					type.setAmount(a.intValue());
-					return type;
-				} else {
-					final EntityType t = ((EntityType) o).clone();
-					t.amount = a.intValue();
-					return t;
-				}
+	protected Object[] get(Event e, Object[] source) {
+		Number a = amount.getSingle(e);
+
+		return get(source, o -> {
+			if (a == null)
+				return null;
```

Comment:
Will this check be ran for every source object? If so, should it just be down outside of get?